React/Modules/RCTNativeAnimationManager.m

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTNativeAnimationManager.h"
+#import "RCTAnimation.h"
+#import <UIKit/UIKit.h>
+
+@implementation RCTNativeAnimationManager {
+  NSMutableDictionary *_animationNodes;
+  NSMutableDictionary *_animationDrivers;
+  NSMutableSet <RCTAnimationDriverNode *> *_activeAnimations;
+  NSMutableSet <RCTAnimationDriverNode *> *_finishedAnimations;
+  NSMutableSet <RCTValueAnimatedNode *> *_updatedValueNodes;
+  NSMutableSet <RCTPropsAnimatedNode *> *_propAnimationNodes;
+  CADisplayLink *displayLink_;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _animationDrivers = [NSMutableDictionary new];
+    _animationNodes = [NSMutableDictionary new];
+    _finishedAnimations = [NSMutableSet new];
+    _propAnimationNodes = [NSMutableSet new];
+    _activeAnimations = [NSMutableSet new];
+    _updatedValueNodes = [NSMutableSet new];
+  }
+  return self;
+}
+
++ (instancetype)sharedManager {
```

Comment:
Why is this a singleton. It seems like it's only used by the RCTNativeAnimatedModule, so couldn't it be part of that module, or at least a property of it?